lib/bank_account/checking_account.rb

```diff
@@ -0,0 +1,44 @@
+module BankAccount
+  class CheckingAccount < Account
+  attr_accessor :check_count, :balance, :id
+
+    def initialize(id, initial_balance)
+      if initial_balance < 0
+        raise ArgumentError.new("You cannot open an account with a negative balance.")
+      else
+      super
+      @check_count = 3
+      end
+    end
+
```

Comment:
Watch your indentation in the two methods below - you want to keep the indentation in between the `else` and `end` consistent with the indentation you see between the `if` and `end`.